robustirc/src/jepsen/robustirc.clj

```diff
@@ -0,0 +1,230 @@
+                "/sbin/start-stop-daemon --start --background --exec ~/gocode/bin/robustirc --"
+                " -listen=" (name node) ":13001"
+                " -network_password=secret"
+                " -network_name=jepsen"
+                " -tls_cert_path=/tmp/cert.pem"
+                " -tls_ca_file=/tmp/cert.pem"
+                " -tls_key_path=/tmp/key.pem"
+                " -singlenode")]
+          (if-not (= node (jepsen/primary test))
+                (Thread/sleep 1000)
+                (do
+                  (info node (str "running: " cmd))
+                  (c/ssh* {:cmd cmd})
+                  (Thread/sleep 5000))))
+        (jepsen/synchronize test)
+        (let [cmd (str
+                "/sbin/start-stop-daemon --start --background --exec ~/gocode/bin/robustirc --"
+                " -listen=" (name node) ":13001"
+                " -network_password=secret"
+                " -network_name=jepsen"
+                " -tls_cert_path=/tmp/cert.pem"
+                " -tls_ca_file=/tmp/cert.pem"
+                " -tls_key_path=/tmp/key.pem"
+                " -join=" (name (jepsen/primary test)) ":13001")]
+          (if (= node (jepsen/primary test))
+                (Thread/sleep 100)
+                (do
+                  (info node (str "running: " cmd))
+                  (c/ssh* {:cmd cmd})
+                  (Thread/sleep 5000))))
+        (jepsen/synchronize test)
+        (info node "setup done")))
+
+    (teardown! [this test node]
+      (c/su
+        (c/ssh* {:cmd "killall robustirc"}))
+      )))
+
+(defn with-nemesis
+  "Wraps a client generator in a nemesis that induces failures and eventually
+  stops."
+  [client]
+  (gen/phases
+    (gen/phases
+      (->> client
+           (gen/nemesis
+             (gen/seq (cycle [(gen/sleep 0)
+                              {:type :info, :f :start}
+                              (gen/sleep 10)
+                              {:type :info, :f :stop}])))
+           (gen/time-limit 30))
+      (gen/nemesis (gen/once {:type :info, :f :stop}))
+      (gen/sleep 5))))
+
+(defn createsession
+  [node]
+  (assoc (json/parse-string
+    (get
+      (httpclient/post (str "https://" (name node) ":13001/robustirc/v1/session")
+        {:insecure? true})
+      :body))
+    :node node))
+
+(defn postmessage
+  [session ircmessage]
+  (let [msgid (bit-or (rand-int Integer/MAX_VALUE)
+                      (Long/parseLong (subs (digest/md5 ircmessage) 17) 16))]
+    (httpclient/post
+                (str "https://" (name (get session :node)) ":13001/robustirc/v1/" (get session "Sessionid") "/message")
+      {:headers {"X-Session-Auth" (get session "Sessionauth")}
+       :insecure? true
+       :content-type :json
+       :form-params {:Data ircmessage
+                     :ClientMessageId msgid}})))
+
+(defn getmessages
+  [session resume]
+  (let [c (async/chan)]
+    (do
+      (async/go
+	; TODO: handle org.apache.http.ConnectionClosedException which gets
```

Comment:
But I don't mind the core.async version either, just figured it might be simpler :)